test/Microsoft.Dnx.Tooling.FunctionalTests/DnuPackTests.cs

```diff
@@ -73,6 +80,8 @@ public void P2PDifferentFrameworks(DnxSdk sdk)
 
             // Assert
             Assert.Equal(0, result.ExitCode);
+
+            TestUtils.CleanUpTestDir<DnuPackTests>(sdk);
```

Comment:
I get the point that it is suppose to clean the folder only if test failed. However it introduces another issue, it the `Theory` case it _may_ clean the test folder anyway because the folder name is decided by __test method name__ + __sdk name__, if there is another factor in the data set (like this case: https://github.com/aspnet/dnx/blob/dev/test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs#L52) then even failed test's record will be erased. 

The point I want to make here is:
1. Function like retaining test trace should be built into the test infrastructure rather than required the test to explicitly invoke it.
2. It's a unstable and weird implementation.